src/c++/perf_analyzer/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -0,0 +1,1149 @@
+        "Populating internal variables");
+    FAIL_IF_ERR(
+        GetSingleton()->LoadServerLibrary(), "Loading Triton Server library");
+    FAIL_IF_ERR(
+        GetSingleton()->StartTriton(memory_type), "Starting Triton Server");
+  }
+
+  return Error::Success;
+}
+
+Error
+TritonLoader::Delete()
+{
+  if (GetSingleton()->server_ != nullptr) {
+    GetSingleton()->server_is_ready_ = false;
+    GetSingleton()->model_is_loaded_ = false;
+    (GetSingleton()->server_).reset();
+  }
+  return Error::Success;
+}
+
+Error
+TritonLoader::PopulateInternals(
+    const std::string& triton_server_path,
+    const std::string& model_repository_path, const std::string& memory_type,
+    bool verbose)
+{
+  RETURN_IF_ERROR(FolderExists(triton_server_path));
+  RETURN_IF_ERROR(FolderExists(model_repository_path));
+  GetSingleton()->triton_server_path_ = triton_server_path;
+  GetSingleton()->model_repository_path_ = model_repository_path;
+  GetSingleton()->verbose_ = verbose;
+  GetSingleton()->verbose_level_ = GetSingleton()->verbose_ ? 1 : 0;
+  return Error::Success;
+}
+
+Error
+TritonLoader::StartTriton(const std::string& memory_type)
+{
+  // Check API version.
+  uint32_t api_version_major, api_version_minor;
+  REPORT_TRITONSERVER_ERROR(
+      GetSingleton()->api_version_fn_(&api_version_major, &api_version_minor));
+  if ((TRITONSERVER_API_VERSION_MAJOR != api_version_major) ||
+      (TRITONSERVER_API_VERSION_MINOR > api_version_minor)) {
+    return Error("triton server API version mismatch");
+  }
+  // Create the server...
+  TRITONSERVER_ServerOptions* server_options = nullptr;
+  RETURN_IF_TRITONSERVER_ERROR(
+      GetSingleton()->options_new_fn_(&server_options),
+      "creating server options");
+  RETURN_IF_TRITONSERVER_ERROR(
+      GetSingleton()->options_set_model_repo_path_fn_(
+          server_options, GetSingleton()->model_repository_path_.c_str()),
+      "setting model repository path");
+  RETURN_IF_TRITONSERVER_ERROR(
+      GetSingleton()->set_log_verbose_fn_(
+          server_options, GetSingleton()->verbose_level_),
+      "setting verbose logging level");
+  RETURN_IF_TRITONSERVER_ERROR(
+      GetSingleton()->set_log_info_fn_(
+          server_options, GetSingleton()->verbose_),
+      "setting if log verbose level is true");
+  RETURN_IF_TRITONSERVER_ERROR(
+      GetSingleton()->set_backend_directory_fn_(
+          server_options,
+          (GetSingleton()->triton_server_path_ + "/backends").c_str()),
+      "setting backend directory");
+  RETURN_IF_TRITONSERVER_ERROR(
+      GetSingleton()->set_repo_agent_directory_fn_(
+          server_options,
+          (GetSingleton()->triton_server_path_ + "/repoagents").c_str()),
+      "setting repository agent directory");
+  RETURN_IF_TRITONSERVER_ERROR(
+      GetSingleton()->set_strict_model_config_fn_(server_options, true),
+      "setting strict model configuration");
+  double min_compute_capability = 0;
+  // FIXME: Do not have GPU support right now
```

Comment:
Do you have follow-up tickets for these enhancements? What is blocking the GPU support on server?